File: GoBuy-fronted/src/api/sku.ts
import request from './request'

export interface Sku {
  id: number
  productId: number
  name: string
  specValues: string
  price: number
  stock: number
  image: string
  createdAt: string
  updatedAt: string
}

export function getSkusByProductId(productId: number): Promise<Sku[]> {
  return request.get(`/skus/product/${productId}`) as Promise<Sku[]>
}

File: GoBuy/src/main/java/com/example/gobuy/modules/product/entity/Sku.java
package com.example.gobuy.modules.product.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.example.gobuy.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("skus")
public class Sku extends BaseEntity {

    private Long productId;

    private String name;

    private String specValues;

    private BigDecimal price;

    private Integer stock;

    private String image;
}
